cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -242,31 +272,40 @@ private void moveReplicas() {
       }
       // After the partition movement finishes, wait for the controller to clean the reassignment zkPath. This also
       // ensures a clean stop when the execution is stopped in the middle.
-      while (!_executionTaskManager.inExecutionTasks().isEmpty()) {
-        LOG.info("Waiting for {} tasks to finish: {}", _executionTaskManager.inExecutionTasks().size(),
-                 _executionTaskManager.inExecutionTasks());
+      Set<ExecutionTask> inExecutionTasks = _executionTaskManager.inExecutionTasks();
+      while (!inExecutionTasks.isEmpty()) {
+        LOG.info("Waiting for {} tasks moving {} MB to finish: {}",
+                 inExecutionTasks.size(), _executionTaskManager.inExecutionDataToMoveInMB(), inExecutionTasks);
         waitForExecutionTaskToFinish();
+        inExecutionTasks = _executionTaskManager.inExecutionTasks();
       }
       if (_executionTaskManager.inProgressTasks().isEmpty()) {
         LOG.info("Partition movements finished.");
-      } else if (_stopRequested) {
-        LOG.info("Partition movements stopped. {} in progress, {} pending, {} aborted, {} dead.",
-                 _executionTaskManager.inProgressTasks().size(),
-                 _executionTaskManager.remainingPartitionMovements().size(),
-                 _executionTaskManager.abortedTasks().size(),
-                 _executionTaskManager.deadTasks().size());
+      } else if (_stopRequested.get()) {
```

Comment:
What if the _executionTaskManager does not have more inProgressTasks, but stopRequested == true? It looks we are going to start to move the leaders in that case?

I think we need to updateOngoingExecutionState in the following places:
1. Within waitForExecutionTaskToFinish();
2. Each time after waitForExecutionTaskToFinish() returns;
3. After moveLeaderships() and moveReplicas();